### ISSUE TITLE
not connecting to Last.fm 

### ISSUE THREAD

**Issue by pciavald** [gnumdk/lollypop#1106]
## Environement
- Lollypop version: 0.9.239
- GTK+ versions: 2.24.31 / 3.22.15
- GNU/Linux distribution: Manjaro 17.0.1 / KDE 5.9.5 / Linux 4.9.30

### Bug/Feature
last.fm refuses to connect. Just tried my login in Last.fm to be sure, I just get this symbol.
![screenshot_20170604_164940](https://cloud.githubusercontent.com/assets/3384097/26762637/fb345ba2-4945-11e7-892d-3b228d7b1469.png)

**Comment by gnumdk** [comment]
What is pylast version?

**Comment by mrnhmath** [comment]
do you have gnome-keyring installed?

**Comment by pciavald** [comment]
pylast version : 1.8.0
I don't have gnome-keyring installed as I'm running KDE. Should it be installed and won't it conflict with KWallet ?

**Comment by gnumdk** [comment]
Do kde keyring implement freedesktop secret? Do you find lastfm password in Kwallet?

**Comment by gnumdk** [comment]
org.gnome.Lollypop.lastfm.login lollypop_gnome

**Comment by gnumdk** [comment]
Don't know if always True but password storage on KDE is quite broken: https://mail.kde.org/pipermail/plasma-devel/2016-July/055668.html

Please install gnome-keyring, if KDE does not provide Freedesktop secret API, it should not conflicts.

**Comment by pciavald** [comment]
i've installed `gnome-keyring`, I have a different behavior but the same outcome. 
When I first tried to log in after installing the package, it asked me to define a gnome-keyring password, so the link is correctly made. I still have the unhappy-face logo, but now there is a little time (~.8s) where the `connect` button remains activated, so something happens where nothing happened before installing gnome-keyring. Before installing when I clicked it was instantaneously failing.

**Comment by gnumdk** [comment]
No proxy?

**Comment by pciavald** [comment]
No proxy, i'm on ethernet to my local router, which is direct house-grade internet

**Comment by pciavald** [comment]
any way to get a verbose of the error ?

**Comment by gnumdk** [comment]
Run this in a python3 console and paste output:

```
from pylast import LastFMNetwork, LibreFMNetwork, md5, BadAuthenticationError
from pylast import SessionKeyGenerator

API_KEY = "7a9619a850ccf7377c46cf233c51e3c6"
API_SECRET = "9254319364d73bec6c59ace485a95c98"

lastfm=LastFMNetwork(api_key=API_KEY, api_secret=API_SECRET)
skg = SessionKeyGenerator(lastfm)
skg.get_session_key(username=login, password_hash=md5(password))
```

**Comment by pciavald** [comment]
not sure if i'm doing this right, but here it is : 
```
~ » python3
Python 3.6.1 (default, Mar 27 2017, 00:27:06) 
[GCC 6.3.1 20170306] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pylast import LastFMNetwork, LibreFMNetwork, md5, BadAuthenticationError
>>> from pylast import SessionKeyGenerator
>>> 
>>> API_KEY = "7a9619a850ccf7377c46cf233c51e3c6"
>>> API_SECRET = "9254319364d73bec6c59ace485a95c98"
>>> 
>>> login="ciavaldp"
>>> password="<obfuscated>"
>>> 
>>> lastfm=LastFMNetwork(api_key=API_KEY, api_secret=API_SECRET)
>>> skg = SessionKeyGenerator(lastfm)
>>> skg.get_session_key(username=login, password_hash=md5(password))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/site-packages/pylast/__init__.py", line 1353, in get_session_key
    doc = request.execute()
  File "/usr/lib/python3.6/site-packages/pylast/__init__.py", line 1233, in execute
    response = self._download_response()
  File "/usr/lib/python3.6/site-packages/pylast/__init__.py", line 1224, in _download_response
    self._check_response_for_errors(response_text)
  File "/usr/lib/python3.6/site-packages/pylast/__init__.py", line 1253, in _check_response_for_errors
    raise WSError(self.network, status, details)
pylast.WSError: Authentication Failed - You do not have permissions to access the service
```

**Comment by gnumdk** [comment]
Here, I get this when entering a wrong password...

**Comment by gnumdk** [comment]
But, search in issues, sure it already happened to someone

**Comment by pciavald** [comment]
i've catched the error : the wrong password put me on the track.
I was sure my password was correct, so I tried to change my password on last.fm. Doing this I realized that my current (very old) password was not in the conditions of the new passwords (at least one special character). I set up a new password following the requirements, now it works. So Last.fm's API must be blocking old passwords, this might be worth to mention on the login page.
